include/arch/x86/ia32/linker.ld

```diff
@@ -533,6 +539,13 @@ SECTIONS
 
 	/DISCARD/ : { *(.note.GNU-stack) }
 
+#ifdef CONFIG_LLVM_USE_BUILTIN_LLD
+	/DISCARD/ : { *(.eh_frame) }
```

Comment:
why do we need to discard the `.eh_frame` for lld ?

If C++ code is compiled with exceptions, then I believe the `.eh_frame` section should be kept.
Else exceptions should probably be disabled with `CONFIG_EXCEPTIONS=n`
https://github.com/zephyrproject-rtos/zephyr/blob/280309e636cb728339b01b52d12b7e2f52216fc7/CMakeLists.txt#L232-L234

Reply:
even without CONFIG_EXCEPTIONS, gun ld and llvm lld they all will generate eh_frame section, but for ld, it will be removed using "-gc-sections" as it is unused, however, for lld, it is still saved even with "-gc-sections", if we don't handle it in linker script, it will report a warning.
so discard it explicitly, because if without CONFIG_EXCEPTIONS, we don't need this section actually, if with CONFIG_EXCEPTIONS, it will be handled by cplusplus-rom.ld before.